services/ui-src/src/utils/other/time.ts

```diff
@@ -14,6 +14,24 @@ export const noon: TimeShape = {
   second: 0,
 };
 
+/*
+ * Defines the beginning and end of Period 1 and Period 2.
+ *     Period 1 is from 01/01 to 06/30.
+ *     Period 2 is from 07/01 to 12/31.
+ */
+export const beginningPeriodOne = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 0, 1);
+};
+export const endPeriodOne = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 5, 30);
+};
+export const beginningPeriodTwo = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 6, 1);
+};
+export const endPeriodTwo = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 11, 31);
+};
+
```

Comment:
I bet you could actually reduce this all down to something like 
```suggestion
return Math.ceil((new Date(date).getMonth() + 1) / 6);
```
That would return the period of either 1 or 2 based on the current month.
The way you've written this though is definitely a lot more readable though so its totally up to you!

Reply:
This is really simple! Thanks for the suggestion! I will implement this. I think the comments are clear enough to understand what the period date ranges are